cmd/webhook/main.go

```diff
@@ -37,12 +42,47 @@ func main() {
 	webhook.SetupInClusterClient()
 
 	/* register handlers */
+	var httpServer *http.Server
 	http.HandleFunc("/validate", webhook.ValidateHandler)
 	http.HandleFunc("/isolate", webhook.IsolateHandler)
 
-	/* start serving */
-	err := http.ListenAndServeTLS(fmt.Sprintf("%s:%d", *address, *port), *cert, *key, nil)
-	if err != nil {
-		glog.Fatalf("error starting web server: %s", err.Error())
+	/* start webhook server */
+	go func() {
+		for {
+			httpServer = &http.Server{
+				Addr: fmt.Sprintf("%s:%d", *address, *port),
+			}
+			err := httpServer.ListenAndServeTLS(*cert, *key)
```

Comment:
This might make you swallow TERM signals, making rolling out changes take much longer.

Reply:
Okey, how about to quit the admission controller instead of restart httpd go routine?
Is that make sense?